File: engine/src/main/java/org/epdb/engine/dto/IntValue.java
package org.epdb.engine.dto;

public record IntValue(int value) implements ColumnValue {

    @Override
    public int comparesTo(ColumnValue other) {
        if(other instanceof IntValue(int otherInt)) {
            return Integer.compare(this.value, otherInt);
        }

        throw new ClassCastException("Cannot compare IntValue with " + other.getClass().getSimpleName());
    }
}

File: modules/engine/src/main/kotlin/org/epdb/engine/columntypes/IntValue.kt
package org.epdb.engine.columntypes

data class IntValue(val value: Int) : ColumnValue {

    override fun comparesTo(other: ColumnValue): Int {
        return when (other) {
            is IntValue -> {
                this.value.compareTo(other.value)
            }

            else -> throw ClassCastException("Cannot compare IntValue with ${other.javaClass.getSimpleName()}")
        }
    }
}